Gems/Multiplayer/Code/Source/Components/SimplePlayerSpawnerComponent.cpp

```diff
@@ -72,35 +74,69 @@ namespace Multiplayer
         incompatible.push_back(AZ_CRC_CE("MultiplayerSpawnerService"));
     }
 
-    NetworkEntityHandle SimplePlayerSpawnerComponent::OnPlayerJoin(
-        [[maybe_unused]] uint64_t userId, [[maybe_unused]] const MultiplayerAgentDatum& agentDatum)
+    AZ::Transform SimplePlayerSpawnerComponent::GetNextSpawnPoint() const
     {
-        const PrefabEntityId prefabEntityId(AZ::Name(m_playerSpawnable.m_spawnableAsset.GetHint().c_str()));
-        AZ::Transform transform = AZ::Transform::Identity();
+        if (m_spawnPoints.empty())
+        {
+            return AZ::Transform::Identity();
+        }
 
+        if (m_spawnIndex >= m_spawnPoints.size())
+        {
+            AZ_Assert(false, "SimplePlayerSpawnerComponent has an out-of-bounds m_spawnIndex %i. Please ensure spawn index is always valid.", m_spawnIndex);
+            return AZ::Transform::Identity();
+        }
+
+        const AZ::EntityId spawnPointEntityId = m_spawnPoints[m_spawnIndex];
+        AZ::Transform spawnPointTransform;
+        AZ::TransformBus::EventResult(spawnPointTransform, spawnPointEntityId, &AZ::TransformInterface::GetWorldTM);
+        return spawnPointTransform;
+    }
+
+    const AZStd::vector<AZ::EntityId>& SimplePlayerSpawnerComponent::GetSpawnPoints() const
+    {
+        return m_spawnPoints;
+    }
+
+    uint32_t SimplePlayerSpawnerComponent::GetSpawnPointCount() const
+    {
+        return aznumeric_cast<uint32_t>(m_spawnPoints.size());
+    }
+
+    uint32_t SimplePlayerSpawnerComponent::GetNextSpawnPointIndex() const
+    {
         if (m_spawnPoints.empty())
         {
-            AZLOG_WARN("SimplePlayerSpawnerComponent is missing spawn points. Spawning new player at the origin.")
+            return static_cast<uint32_t>(-1);
         }
-        else
+
+        if (m_spawnIndex >= m_spawnPoints.size())
         {
-            m_spawnIndex %= m_spawnPoints.size();
-            const AZ::EntityId spawnPointEntityId = m_spawnPoints[m_spawnIndex];
-            
-            AZ::Entity* spawnPointEntity = nullptr;
-            AZ::ComponentApplicationBus::BroadcastResult(
-                spawnPointEntity, &AZ::ComponentApplicationBus::Events::FindEntity, spawnPointEntityId);
-            if (spawnPointEntity != nullptr)
-            {
-                transform = spawnPointEntity->GetTransform()->GetWorldTM();
-            }
-            else
-            {
-                AZLOG_WARN("The spawn point entity id for index %i is invalid. Spawning new player at the origin.", m_spawnIndex)
-            }
+            AZ_Assert(false, "SimplePlayerSpawnerComponent has an out-of-bounds m_spawnIndex %i. Please ensure spawn index is always valid.", m_spawnIndex);
+            return static_cast<uint32_t>(-1);
```

Comment:
Suggestion: I think for an empty list it would be more reasonable to return an index of 0 instead of 4 billion. 0 would still be easily detectable as invalid because it would be >= GetSpawnPointCount().

Reply:
Updated!